spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+var languageNegotiationLambda = require( '../../src/language_negotiation_lambda' );
+
+describe( 'languageNegotiationLambda', function() {
+    var event,
+        default_header = ['en'];
+
+    beforeEach(function(){
+        // Cloudfront Request Object
+        event = {
+            "Records": [
+                {
+                  "cf": {
+                    "request": {
+                      "headers": {
+                        "Host": [
+                          "localhost"
+                        ],
+                        "User-Agent": [
+                          "Test Agent"
+                        ]
+                      },
+                      "clientIp": "2001:cdba::3257:9652",
+                      "httpVersion": "2.0",
+                      "uri": "/",
+                      "method": "GET"
+                    },
+                    "config": {
+                      "distributionId": "EXAMPLE"
+                    }
+                  }
+                }
+            ]
+        };
+    });
+
+    describe( 'Valid cookies', function(){
+        [
+            { cookie: "en",     expected: "en" },
+            { cookie: "EN",     expected: "en" },
+            { cookie: "es",     expected: "es" },
+            { cookie: "ES",     expected: "es" },
+            { cookie: "en-US",  expected: "en" },
+            { cookie: "es-419", expected: "es" },
+            { cookie: "es_419", expected: "es" },
+            { cookie: "fr",     expected: "es" },
+            { cookie: "",       expected: "es" }
```

Comment:
I would argue that this is testing "valid" cookies.  Of which, empty is valid and has an expected result.  The `es` thing is related to your above comment, for which I am open to suggestions